llvm/include/llvm/MC/MCRepoObjectWriter.h

```diff
@@ -57,15 +53,14 @@ struct RepoRelocationEntry {
 };
 
 class MCRepoObjectTargetWriter : public MCObjectTargetWriter {
-  virtual void anchor();
```

Comment:
The anchor() implementation is removed here, but it is retained in the X86RepoMCAsmInfo class. That seems inconsistent. Is there a reason?

Reply:
>
> Is there a reason?
>

I read the LLVM code standards of [provide a virtual method anchor for classes in headers](https://releases.llvm.org/2.7/docs/CodingStandards.html#ll_virtual_anch). 

The `MCRepoObjectTargetWriter` class already includes several virtual functions, e.g.` getRelocType`, `needsRelocateWithSymbol` and `sortRelocs`. It doesn’t need to add virtual `anchor()` function to avoid to copy the vtable and RTTI into every .o file that #includes the header. We can see that classes `MCELFObjectTargetWriter`, `MCMachObjectTargetWriter` and ` MCWasmObjectTargetWriter`don’t have the virtual `anchor` function as well.

The `X86RepoMCAsmInfo` class does not have virtual functions but it is derived from ` MCAsmInfoRepo`, which has virtual functions. Therefore, It needs to add `anchor()` function to avoid to copy the vtable and RTTI into every .o file that #includes the header. Again, we can see that classes ` MCAsmInfoMicrosoft`, `MCAsmInfoCOFF`, ` MCAsmInfoELF` etc. include function `anchor’ definition as well.